cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -705,6 +705,14 @@ struct BinaryScalarMinMax {
 template <typename Op>
 struct FixedSizeBinaryScalarMinMax {
   static Status Exec(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
+    if (batch.IsNull()) {
+      std::shared_ptr<Scalar> result = MakeNullScalar(out->type()->GetSharedPtr());
+      ARROW_ASSIGN_OR_RAISE(std::shared_ptr<Array> arr_result,
+                            MakeArrayFromScalar(*result, 1));
+      out->value = std::move(arr_result->data());
```

Comment:
This factory allocates and so should use the pool provided by the kernel context. Also this can be written more concisely with MakeArrayOfNull
```suggestion
      ARROW_ASSIGN_OR_RAISE(out->value,
                            MakeArrayOfNull(out->type()->GetSharedPtr(), 1
                                            ctx->memory_pool()));
```